src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -1044,6 +1054,9 @@ string notEquals(Field field)
         private static bool TypeIsSymbol(Field field) => TypeIsSymbol(field.Type);
         private static bool TypeIsSymbol(string type) => type.TrimEnd('?').EndsWith("Symbol");
 
+        private static bool TypeIsNonTypeSymbol(Field field) => TypeIsNonTypeSymbol(field.Type);
+        private static bool TypeIsNonTypeSymbol(string type) => TypeIsSymbol(type) && !type.TrimEnd('?').EndsWith("TypeSymbol");
```

Comment:
>EndsWith("TypeSymbol") [](http://example.com/codeflow?start=105&length=22)

nit: Not all types deriving from TypeSymbol end with "TypeSymbol", but the ones we currently use in BoundNodes.xml are a limited subset which do follow that naming convention. 
Consider asserting that if the name contains "Type" and "Symbol" then it does end in "TypeSymbol". That will increase the chances of catching that a change is needed here if we ever end up using one of those in BoundNodes.xml